Making the use of a pool optional for ExecuteFetch

alainjobart opened the pull request:
ExecuteFetchAsDba and ExecuteFetchAsApp, part of tabletmanager client
API, now have a flag to make the use of a connection pool optional. It's
only enabled in vtworker, the only client that really needs the speed
gain.

@michael-berlin @enisoc

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/youtube/vitess/1906)
<!-- Reviewable:end -->

enisoc commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> overall, but one concern.

---

Reviewed 12 of 12 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[go/vt/wrangler/schema.go, line 339 \[r1\]](https://reviewable.io:443/reviews/youtube/vitess/1906#-KNsiuJ2fGech-Mpdb8s:-KNsiuJ2fGech-Mpdb8t:1626867796) ([raw file](https://github.com/youtube/vitess/blob/adb60ca6ca54baf106f504c92df00b488c0e476b/go/vt/wrangler/schema.go#L339)):*
> ```Go
> 	defer cancel()
> 	// Need to make sure that we enable binlog, since we're only applying the statement on masters.
> 	_, err = wr.tmc.ExecuteFetchAsDba(ctx, tabletInfo.Tablet, true, []byte(filledChange), 0, false, reloadSchema)
> ```

Why does this one need the pool? If vtctld calls this, the pool of connections will never be closed, right?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/youtube/vitess/1906#-:-KNsjR3lylzuEMSfqFAZ:88033593)*
<!-- Sent from Reviewable.io -->

alainjobart commented:
Addressed the concern.

---

Review status: 11 of 12 files reviewed at latest revision, 1 unresolved discussion.

---

*[go/vt/wrangler/schema.go, line 339 \[r1\]](https://reviewable.io:443/reviews/youtube/vitess/1906#-KNsiuJ2fGech-Mpdb8s:-KNswTmtTlcMieEabvp3:1928133489) ([raw file](https://github.com/youtube/vitess/blob/adb60ca6ca54baf106f504c92df00b488c0e476b/go/vt/wrangler/schema.go#L339)):*
<details><summary><i>Previously, enisoc (Anthony Yeh) wrote…</i></summary>
> Why does this one need the pool? If vtctld calls this, the pool of connections will never be closed, right?
</details>
we call Close() now on the connections, so it would be closed. However, this is a bit useless, so I changed it anyway.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/youtube/vitess/1906#-:-KNsweF8tMx_NTICAW2L:108847919)*
<!-- Sent from Reviewable.io -->
